src/models/model.js

```diff
@@ -83,22 +80,87 @@ export default class Model {
 
     // If it's a partial update, get only the fields specified
     if (fields.length > 0) {
-      fields.forEach(field => {
+      fields.forEach((field) => {
         data[field] = obj[field]
       })
     } else {
       data = { ...obj }
     }
 
     // Delete private '_fields' member
-    delete data['_fields'];
-
+    delete data['_fields']
+    // format it the way it is expected
+    Object.keys(data).forEach((k) => {
+      if (
+        data[k] instanceof Model ||
+        (Array.isArray(data[k]) && data[k].length && data[k][0] instanceof Model)
+      ) {
+        if (Array.isArray(data[k])) {
+          data[k] = data[k].map((value) => value.constructor.toAPI(value))
+        } else data[k] = data[k].constructor.toAPI(data[k])
```

Comment:
Personal preference perhaps, but I'd write this as a ternary assignment, ex:

```
data[k] = Array.isArray(data[k])
    ? data[k].map((value) => value.constructor.toAPI(value))
    : data[k] = data[k].constructor.toAPI(data[k])
```

I feel like it's a bit easier to read.